client/my-sites/email/email-management/home/email-plan-view.jsx

```diff
@@ -0,0 +1,220 @@
+
+const normalizeTitanMailboxes = ( titanMailboxes ) => {
+	if ( ! titanMailboxes ) {
+		return [];
+	}
+	return titanMailboxes.map( ( mailbox ) => {
+		return {
+			email: mailbox.email,
+			isAdmin: mailbox.is_admin,
+		};
+	} );
+};
+
+class EmailPlanView extends React.Component {
+	state = {
+		isLoadingTitanMailboxes: false,
+		errorLoadingTitanMailboxes: false,
+		loadedTitanMailboxes: false,
+		titanMailboxes: [],
+	};
+
+	componentDidMount() {
+		this.loadTitanMailboxes();
+	}
+
+	componentDidUpdate() {
+		this.loadTitanMailboxes();
+	}
+
+	loadTitanMailboxes() {
+		const { selectedSite } = this.props;
+
+		if ( this.state.isLoadingTitanMailboxes || this.state.loadedTitanMailboxes ) {
+			return;
+		}
+
+		this.setState( {
+			isLoadingTitanMailboxes: true,
+		} );
+
+		wp.undocumented()
+			.getTitanMailboxesForSite( selectedSite.ID )
+			.then(
+				( data ) => {
+					this.setState( {
+						isLoadingTitanMailboxes: false,
+						errorLoadingTitanMailboxes: false,
+						loadedTitanMailboxes: true,
+						titanMailboxes: data?.accounts || [],
+					} );
+				},
+				() => {
+					this.setState( {
+						isLoadingTitanMailboxes: false,
+						errorLoadingTitanMailboxes: true,
+						loadedTitanMailboxes: true,
+						titanMailboxes: [],
+					} );
+				}
+			);
+	}
+
+	getEmails() {
+		const { domain, emails } = this.props;
+
+		if ( hasTitanMailWithUs( domain ) ) {
+			return normalizeTitanMailboxes( this.state.titanMailboxes );
+		}
+
+		return emails;
+	}
+
+	handleBack = () => {
+		const { selectedSite } = this.props;
+		page( emailManagement( selectedSite.slug ) );
+	};
+
+	render() {
+		const {
+			domain,
+			selectedSite,
+			hasEmailPlanSubscription,
+			purchase,
+			isLoadingPurchase,
+		} = this.props;
+
+		const { statusClass, text, icon } = resolveEmailPlanStatus( domain );
+
+		const cardClasses = classnames( 'email-plan-view__general', statusClass );
+
+		return (
+			<React.Fragment>
+				{ domain && <QueryEmailForwards domainName={ domain.name } /> }
+				{ selectedSite && hasEmailPlanSubscription && (
+					<QuerySitePurchases siteId={ selectedSite.ID } />
+				) }
+				<HeaderCake onClick={ this.handleBack }>Email plan settings</HeaderCake>
+				<CompactCard className={ cardClasses }>
+					<span className="email-plan-view__general-icon">
+						<EmailTypeIcon domain={ domain } />
+					</span>
+					<div>
+						<h2>@{ domain.name }</h2>
+						<span className="email-plan-view__status">
+							<MaterialIcon icon={ icon } /> { text }
+						</span>
+					</div>
+				</CompactCard>
+
+				{ hasEmailPlanSubscription && (
+					<EmailPlanSubscription
+						purchase={ purchase }
+						domain={ domain }
+						selectedSite={ selectedSite }
+						isLoadingPurchase={ isLoadingPurchase }
+					/>
+				) }
+
+				<EmailPlanMailboxesList emails={ this.getEmails() } />
+
+				<VerticalNav>
+					<VerticalNavItem path="a">Add new mailbox</VerticalNavItem>
+				</VerticalNav>
+			</React.Fragment>
+		);
+	}
+}
```

Comment:
I would recommend refactoring this component. At first I thought it would handle all types of email accounts. Then I started reading it, and found that it actually contains lots of logic that is specific to Titan. And finally, I've realized that it's indeed for G Suite and email forwards when I reached the part where we connect it to the Redux state tree. We should really separate responsibilities, extract the code that retrieves the list of Titan mailboxes in a Query component, moves the normalization logic in another component ...

Reply:
The code is in there temporary - I just didn't want to add all the boilerplate code in one single PR as that would make the PR even bigger. This is actually on my list of next tasks - "Move the Titan mailboxes loading into global state"

Reply:
Gotcha. I've mentioned it in the server patch but I believe we could simplify the code a lot if data normalization happened on the server, and if we used one API endpoint to retrieve all the email accounts (may they be for Titan or Google Workspace).

Reply:
Actually there is a stub for this called `QueryEmailAccounts` and related redux stuff in `client/state/email-accounts` but it would require a lot more changed to actually make this work so I decided on making this PR work instead and leave the refactoring for later. It would definitely make the code simpler but it's just too big to include in this PR.